lib/models/food.js

```diff
@@ -2,82 +2,80 @@ const environment = process.env.NODE_ENV || 'development';
+      return callback(new Error('Food must have calories.'));
+    }
+    callback();
+  };
 
-    calorieValidation(calories, (err) => {
-      if (err) {
-        return reject(err);
-      }
+  nameValidation (name, callback) {
+    if (!name) {
+      return callback(new Error('Food must have name.'));
+    }
+    callback();
+  }
+
+  all () {
+    return database.raw('SELECT * FROM foods');
+  };
 
-      nameValidation(name, (err) => {
+  create (name, calories) {
+    return new Promise( (resolve, reject) => {
+
+      this.calorieValidation(calories, (err) => {
         if (err) {
           return reject(err);
         }
 
-        database.raw('INSERT INTO foods (name, calories, created_at) VALUES (?, ?, ?) RETURNING id, name, calories',
-          [name, calories, new Date])
-            .then(resolve)
-            .catch(reject)
-      })
-    });
-  })
-}
-function find (id) {
-  return database.raw('SELECT * FROM foods WHERE id = ?', id)
-};
+        this.nameValidation(name, (err) => {
+          if (err) {
+            return reject(err);
+          }
 
-function update(req) {
-  return new Promise( (resolve, reject) => {
-    const food = req.body.food
+          database.raw('INSERT INTO foods (name, calories, created_at) VALUES (?, ?, ?) RETURNING id, name, calories',
+            [name, calories, new Date])
+              .then(resolve)
+              .catch(reject)
+        });
+      });
+    });
+  };
 
-    let query;
+  find (id) {
+    return database.raw('SELECT * FROM foods WHERE id = ?', id)
+  }
 
-    if (!food.calories && food.name) {
-      query = database.raw('UPDATE foods SET name = ?, created_at = ? WHERE id = ?',
-        [food.name, new Date, req.params.id])
-    } else if (food.calories && !food.name) {
-      query = database.raw('UPDATE foods SET calories = ?, created_at = ? WHERE id = ?',
-        [food.calories, new Date, req.params.id])
-    } else {
-      query = database.raw('UPDATE foods SET name = ?, calories = ?, created_at = ? WHERE id = ?',
-        [food.name, food.calories, new Date, req.params.id])
-    }
+  update (req) {
+    return new Promise( (resolve, reject) => {
+      const food = req.body.food
 
-    query.then(() => {
-      return find(req.params.id)
-    })
-    .then(resolve)
-    .catch(reject)
-  });
-}
+      let query;
 
-function destroy(req) {
-  return database.raw('DELETE FROM foods WHERE id = ?', req.params.id)
-}
+      if (!food.calories && food.name) {
+        query = database.raw('UPDATE foods SET name = ?, created_at = ? WHERE id = ?',
+          [food.name, new Date, req.params.id])
+      } else if (food.calories && !food.name) {
+        query = database.raw('UPDATE foods SET calories = ?, created_at = ? WHERE id = ?',
+          [food.calories, new Date, req.params.id])
+      } else {
+        query = database.raw('UPDATE foods SET name = ?, calories = ?, created_at = ? WHERE id = ?',
+          [food.name, food.calories, new Date, req.params.id])
+      }
```

Comment:
@cmacaulay you could extract things a little more so it's more readable - you could have something like 

```js
if(updateNameOnly()) {
//...
} else if (updateCaloriesOnly()) {
//...
} else {
//...
}
```

you could also break down the SQL statements into smaller functions, but i think what you have is okay!